### PR TITLE
Refactor GlowMeta*

### PR DESCRIPTION
**Pull request by Pr0methean**
List getters and setters now consistently make defensive copies.
Non-list getters and setters are lombokified (including a few that were stubs). Copy constructors now
handle *any* implementations of their respective interfaces, improving
plugin-friendliness. GlowMetaShield now extends GlowMetaBanner.
GlowMetaSkull.setOwningPlayer now checks the ProfileCache but still
doesn't wait for a profile to be downloaded. Missing Javadocs are added.

### CODE REVIEW

**Comment by smartboyathome** [comment]
Rather than defensive copies, you should think about wrapping the lists in Collections.unmodifiableList() where possible.

**Comment by Pr0methean** [comment]
@smartboyathome If I wrap a list in Collections.unmodifiableList while there are still other references to it, from which it can be modified, then it won't be thread-safe. Plus, even if we returned a thread-safe read-only view, we'd still run into trouble when people were assuming they were getting a snapshot.

**Review by aramperes — DISMISSED** [review]

File: src/main/java/net/glowstone/inventory/GlowMetaSkull.java [hunk 1]
```diff
@@ -93,24 +120,44 @@ public boolean setOwner(String name) {
...
+        return setOwningPlayerInternal(owningPlayer);
+    }
+
+    private boolean setOwningPlayerInternal(OfflinePlayer owningPlayer) {
         if (owningPlayer instanceof GlowOfflinePlayer) {
             GlowOfflinePlayer impl = (GlowOfflinePlayer) owningPlayer;
-            this.owner = impl.getProfile();
+            this.owner.set(impl.getProfile());
+            return true;
+        } else {
```
Comment (kaenganxt) [inline src/main/java/net/glowstone/inventory/GlowMetaSkull.java]: `owningPlayer` could also be a `GlowPlayer` (which does not extend `GlowOfflinePlayer`).

**Review by kaenganxt — CHANGES_REQUESTED** [review]

File: src/main/java/net/glowstone/inventory/GlowMetaSkull.java (hunk 1 above)
Comment (Pr0methean) [reply src/main/java/net/glowstone/inventory/GlowMetaSkull.java]: Handled that case.

**Review by kaenganxt — APPROVED** [review]

File: src/main/java/net/glowstone/inventory/GlowMetaSkull.java [hunk 2]
```diff
@@ -147,6 +148,8 @@ private boolean setOwningPlayerInternal(OfflinePlayer owningPlayer) {
             GlowOfflinePlayer impl = (GlowOfflinePlayer) owningPlayer;
             this.owner.set(impl.getProfile());
             return true;
+        } else if (owningPlayer instanceof GlowPlayer) {
+            this.owner.set(((GlowPlayer) owningPlayer).getProfile());
```
Comment (aramperes) [inline src/main/java/net/glowstone/inventory/GlowMetaSkull.java]: Missing a return statement, then it should be good to go.

**Review by aramperes — APPROVED** [review]

**Pull request merged**